raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -195,7 +216,7 @@ def handle_routechange(state, state_change):
 
 def handle_transferrefund(state, state_change):
     if state_change.sender == state.route.node_address:
-        iteration = cancel_current_route(state)
+        iteration = cancel_current_route(state, send_reveal_secret=True)
```

Comment:
revealing the secret is only done here, please pull the code from the `cancel_current_route` into this function.